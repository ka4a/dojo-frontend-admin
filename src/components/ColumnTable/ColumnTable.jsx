import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './ColumnTable.scss';

const ColumnTable = ({ data }) => (
  <section>
    {data.map(({ title, content }) => (
      <div className="mb-4" key={title}>
        <h4>{title}</h4>
        <div className="column-table__row">
          {content.map(({ subtitle, value }, index) => (
            <div
              key={`${value}-${subtitle}`}
              className={classNames({
                'border-right border-light-100': content.length - 1 > index,
                'pl-5': index > 0,
              })}
            >
              <span className="small">{subtitle}</span>
              <br />
              <span className="text-gray-500 x-small font-weight-bold">
                {value}
              </span>
            </div>
          ))}
        </div>
      </div>
    ))}
  </section>
);

ColumnTable.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      content: PropTypes.arrayOf(
        PropTypes.shape({
          subtitle: PropTypes.string,
          value: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
        }),
      ),
    }),
  ),
};

ColumnTable.defaultProps = {
  data: [],
};

export default memo(ColumnTable);
