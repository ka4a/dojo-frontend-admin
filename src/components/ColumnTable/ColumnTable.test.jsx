import { shallow, mount } from 'enzyme';
import ColumnTable from './ColumnTable';

const defaultProps = {
  data: [
    {
      title: 'test title',
      content: [{ subtitle: 'test subtitle', value: 'test value' }],
    },
  ],
};

describe('ColumnTable', () => {
  it('it should render ColumnTable component', () => {
    shallow(<ColumnTable {...defaultProps} />);
  });

  it('it should render the expected HTML', () => {
    expect(mount(<ColumnTable {...defaultProps} />).html()).toMatchSnapshot();
  });
});
