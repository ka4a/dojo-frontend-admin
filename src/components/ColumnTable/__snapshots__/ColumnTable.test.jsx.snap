// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnTable it should render the expected HTML 1`] = `"<section><div class=\\"mb-4\\"><h4>test title</h4><div class=\\"column-table__row\\"><div class=\\"\\"><span class=\\"small\\">test subtitle</span><br><span class=\\"text-gray-500 x-small font-weight-bold\\">test value</span></div></div></div></section>"`;
